lib/enum/basic_enum.rb

```diff
@@ -85,66 +98,86 @@ def element_frequency_count
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-find-all
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-select
   def elements_ending_in_er
+    collection.find_all { |element| element.include? "er" }
   end
 
 
   # Refer to find_index
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-find-index
   def index_of_first_awesome_element
+    collection.find_index { |i| i[:awesome] == true }
   end
 
   # Refer to group_by
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-group-by
   def group_elements_by_favorite_language
+    #broken
+    collection.group_by { |item| item[:favorite_language] }
   end
 
   # Refer to inject / reduce
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-inject
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-reduce
   def sum_of_experiences
+    collection.inject(0) { |sum, key| sum + key[:years_experience] }
   end
 
   # Refer to inject / reduce
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-inject
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-reduce
   def longest_element_name_using_inject
+    collection.inject do |memo, hash|
```

Comment:
Yes, and if you wanted to do this as a one-liner, you could use the ternary operator, like this: `collection.inject {|longest, element| longest[:name].length > element[:name].length ? longest : element }`